jupyter_server_proxy/handlers.py

```diff
@@ -373,6 +373,82 @@ async def proxy(self, host, port, proxied_path):
             )
         else:
             client = httpclient.AsyncHTTPClient()
+        # check if the request is stream request
+        proxy_streaming = self.request.headers.get('Accept')
+        if proxy_streaming == 'text/event-stream':
+            return await self._proxy_progressive(host, port, proxied_path, body, client)
+        else:
+            return await self._proxy_buffered(host, port, proxied_path, body, client)
+    
+    async def _proxy_progressive(self, host, port, proxied_path, body, client):
+        # Proxy in progressive flush mode, whenever chunks are received. Potentially slower but get results quicker for voila
+        # Set up handlers so we can progressively flush result
+
+        headers_raw = []
+
+        def dump_headers(headers_raw):
+            for line in headers_raw:
+                r = re.match('^([a-zA-Z0-9\-_]+)\s*\:\s*([^\r\n]+)[\r\n]*$', line)
+                if r:
+                    k,v = r.groups([1,2])
+                    if k not in ('Content-Length', 'Transfer-Encoding',
+                                  'Content-Encoding', 'Connection'):
+                        # some header appear multiple times, eg 'Set-Cookie'
+                        self.set_header(k,v)
+                else:
+                    r = re.match('^HTTP[^\s]* ([0-9]+)', line)
```

Comment:
Same as my comment in https://github.com/jupyterhub/jupyter-server-proxy/pull/479/files#r1645266065

Reply:
here is a example for stream header:
```
HTTP/1.1 200 OK
Content-Type: text/html
Cache-Control: no-cache
Connection: keep-alive
```
httputil.HTTPHeaders.parse_line  can not parse the first line to get http status, so we use regex to solve this issue.
the source code comes from https://github.com/ideonate/jhsingle-native-proxy/blob/3e26a4ee0e7318970b7cf6abbd7d88455a9ac621/jhsingle_native_proxy/proxyhandlers.py#L265